src/rgw/rgw_rados.cc

```diff
@@ -12369,11 +12369,9 @@ int RGWRados::cls_user_sync_bucket_stats(rgw_raw_obj& user_obj, const RGWBucketI
 
   bucket_info.bucket.convert(&entry.bucket);
 
-  map<string, struct rgw_bucket_dir_header>::iterator hiter = headers.begin();
-  for (; hiter != headers.end(); ++hiter) {
-    map<uint8_t, struct rgw_bucket_category_stats>::iterator iter = hiter->second.stats.begin();
-    for (; iter != hiter->second.stats.end(); ++iter) {
-      struct rgw_bucket_category_stats& header_stats = iter->second;
+  for (auto hiter : headers) {
+    for (auto iter : hiter.second.stats) {
+      struct rgw_bucket_category_stats& header_stats = iter.second;
```

Comment:
careful with copies in these for loops. `iter` here is a `pair<uint8_t, rgw_bucket_category_stats>`, rather than a `map<uint8_t, rgw_bucket_category_stats>::iterator`, so you need `const auto& iter` to avoid making a copy